root/user/UserCollections.js

```diff
@@ -0,0 +1,265 @@
+
+type Props = {|
+  +$c: {user: EditorT} & CatalystContextT,
+  +collaborativeCollections: CollectionListT,
+  +ownCollections: CollectionListT,
+  +user: EditorT,
+|};
+
+type CollectionListT = {|
+  +[string]: $ReadOnlyArray<CollectionT>,
+|};
+
+const collectionsListTitle = {
+  area: N_l('Area Collections'),
+  artist: N_l('Artist Collections'),
+  event: N_l('Event Collections'),
+  instrument: N_l('Instrument Collections'),
+  label: N_l('Label Collections'),
+  place: N_l('Place Collections'),
+  recording: N_l('Recording Collections'),
+  release: N_l('Release Collections'),
+  release_group: N_l('Release Group Collections'),
+  series: N_lp('Series Collections', 'plural'),
+  work: N_l('Work Collections'),
+};
+
+const CollectionsTable = (({columns, data}) => {
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+  } = useTable({
+    columns,
+    data,
+  });
+
+  /* eslint-disable react/jsx-key */
+  return (
+    <table {...getTableProps({className: 'tbl'})}>
+      <thead>
+        {headerGroups.map(headerGroup => (
+          <tr {...headerGroup.getHeaderGroupProps()}>
+            {headerGroup.headers.map(column => (
+              <th {...column.getHeaderProps()}>{column.render('Header')}</th>
+            ))}
+          </tr>
+        ))}
+      </thead>
+      <tbody {...getTableBodyProps()}>
+        {rows.map(
+          (row, i) => prepareRow(row) || (
+            <tr {...row.getRowProps({className: loopParity(i)})}>
+              {row.cells.map(cell => (
+                <td {...cell.getCellProps(cell.column.id === 'actions'
+                  ? {className: 'actions'} : {})}
+                >
+                  {cell.render('Cell')}
+                </td>
+              ))}
+            </tr>
+          ),
+        )}
+      </tbody>
+    </table>
+  );
+  /* eslint-enable react/jsx-key */
+});
+
+function formatCollaboratorNumber(
+  collaborators: $ReadOnlyArray<EditorT>,
+  currentUserId: number,
+) {
+  const isCollaborator = some(
+    collaborators,
+    (collaborator => collaborator.id === currentUserId),
+  );
+
+  return isCollaborator ? (
+    exp.l('{collaborator_number} (including you)',
+          {collaborator_number: collaborators.length})
+  ) : (
+    collaborators.length
+  );
+}
+
+function formatPrivacy(isPublic) {
+  return isPublic ? l('Public') : l('Private');
+}
+
+function collaborativePrivacy(
+  collection,
+  currentUser,
+) {
+  return collection.editor.id === currentUser.id ? (
+    formatPrivacy(collection.public) + ' ' + l('(your collection)')
+  ) : (
+    formatPrivacy(collection.public)
+  );
+}
+
+const CollectionsEntitySection = ({
+  $c,
+  isCollaborative,
+  collections,
+  type,
+  user,
+}) => {
+  const viewingOwnProfile = !!$c.user && $c.user.id === user.id;
+  const nameColumn = {
+    Cell: ({row: {original}}) => (
+      <EntityLink entity={original} />
+    ),
+    Header: N_l('Collection'),
+    accessor: 'name',
+  };
```

Comment:
One use I can see of building tables this way is that we can move these column descriptions outside of the component, and export/import them around so we can compose them together to customize which columns are shown in different tables. (Though maybe the `hideFoo` props we have already work just as well.)

Reply:
Exactly, answered in https://github.com/metabrainz/musicbrainz-server/pull/1246#issuecomment-545968391.